### PR TITLE
Enable operator metrics collection

### PR DESCRIPTION
**Pull request by OlivierCazade**
Enable collection for operator metrics.

### CODE REVIEW

File: config/prometheus/monitor.yaml [hunk 1]
```diff
@@ -12,7 +12,6 @@ spec:
     - path: /metrics
       port: https
       scheme: https
```
Comment (jotak) [inline config/prometheus/monitor.yaml:14]: What is missing for having a TLS without skipping verify?
It's basically the same reason why we have this bug: https://issues.redhat.com/browse/NETOBSERV-765

Comment (OlivierCazade) [reply config/prometheus/monitor.yaml:14]: Similar to what we did for FLP we need to add the labels to the metric services to generate certificate and mount them in the kube-rbac-proxy container so they can be used.

**Comment by OlivierCazade** [comment]
Added SSL and auth configuration for downstream collection, this made operator metrics endpoint incompatible with monitoring user workload.

@memodi appart from review comments, the PR is ready to be tested.
After deploying this branch and setting DOWNSTREAM_DEPLOYMENT to true, you should see operator metrics without enabling user workload.

image

**Comment by memodi** [comment]
> Added SSL and auth configuration for downstream collection, this made operator metrics endpoint incompatible with monitoring user workload.

@OlivierCazade - is the idea, they would be compatible with SSL and auth configuration once operator is deployed with DOWNSTREAM_DEPLOYMENT to true?

**Comment by OlivierCazade** [comment]
Yes, SSL and authentication for the metrics endpoint. And the prometheus deployment should automatically use it from the ServiceMonitor configuration

**Comment by memodi** [comment]
@OlivierCazade - I tried with image from this PR, I am not able to see those metrics:
```
$ oc get pod/netobserv-controller-manager-7794cf659b-vcnhn -o yaml | egrep -A 1 "DOWN"
    - --downstream-deployment=$(DOWNSTREAM_DEPLOYMENT)
    command:
--
    - name: DOWNSTREAM_DEPLOYMENT
      value: "true"

$ oc get servicemonitor
NAME                        AGE
netobserv-metrics-monitor   5m19s
```
image

am I missing something?

**Comment by OlivierCazade** [comment]
Most of the changes in this PR are at the bundle level, did you also used the bundle from this branch?

File: monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
```
Comment (jotak) [inline monitor.yaml:1]: I don't think we want this file, do we? Is it included in the bundles, or is it just there as a helper (e.g. for make targets) ? afaict it should not be included in bundles

File: config/openshift-olm/monitor.yaml [hunk 3]
```diff
@@ -9,12 +8,14 @@ metadata:
...
+    - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      path: /metrics
+      interval: 30s
       port: https
       scheme: https
-      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
       tlsConfig:
-        insecureSkipVerify: true
+        caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
+        serverName: netobserv-metrics-service.netobserv.svc
```
Comment (jotak) [inline config/openshift-olm/monitor.yaml:18]: so, my understanding is that we should hardcode here the installation namespace, `openshift-operators`, right?

File: monitor.yaml (hunk 2 above)
Comment (OlivierCazade) [reply monitor.yaml:1]: No sorry, this was one of my test file.

**Review by jotak — APPROVED** [review]

**Comment by memodi** [comment]
adding NETOBSERV-900 as JIRA reference

**Pull request merged**